src/download_fossa_cli.rs

```diff
@@ -58,20 +68,68 @@ pub async fn ensure_fossa_cli() -> Result<PathBuf, Error> {
     let data_root = io::data_root().await.change_context(Error::SetDataRoot)?;
 
     // default to fossa that lives in ~/.config/fossa/broker/fossa
+    // if it does not exist in ~/.config/fossa/broker/fossa, then check to see if it is on the path
     let command_in_config_dir = data_root.join(command);
-    if check_command_existence(&command_in_config_dir).await {
-        return Ok(command_in_config_dir);
+    let current_path: Option<PathBuf> = if check_command_existence(&command_in_config_dir).await {
+        Some(command_in_config_dir)
+    } else if check_command_existence(&PathBuf::from(&command)).await {
+        Some(PathBuf::from(command))
+    } else {
+        None
+    };
+
+    let latest_release_version = latest_release_version().await?;
+    match current_path {
+        Some(current_path) => {
+            download_if_old(data_root, current_path, latest_release_version).await
+        }
+        None => download(data_root, latest_release_version).await,
     }
+}
 
-    // if it does not exist in ~/.config/fossa/broker/fossa, then check to see if it is on the path
-    if check_command_existence(&PathBuf::from(&command)).await {
-        return Ok(PathBuf::from(command));
-    };
+/// If the cli exists locally, then compare the version of the local CLI to the latest release,
+/// and download it if it is different.
+/// If there are any errors while finding the local version, then just download it.
+#[tracing::instrument]
+async fn download_if_old(
+    data_root: &PathBuf,
+    current_path: PathBuf,
+    latest_release_version: String,
+) -> Result<PathBuf, Error> {
+    if let Ok(local_version) = local_version(&current_path).await {
```

Comment:
I really wanted to use [is_okay_and](https://doc.rust-lang.org/std/result/enum.Result.html#method.is_ok_and) here, but it's only a nightly feature :cry: